corehq/apps/users/models.py

```diff
@@ -1655,8 +1655,8 @@ def save(self, fire_signals=True, **params):
             from .signals import commcare_user_post_save
             results = commcare_user_post_save.send_robust(sender='couch_user', couch_user=self,
                                                           is_new_user=is_new_user)
-            sync_user_cases.delay(self)
             log_signal_errors(results, "Error occurred while syncing user (%s)", {'username': self.username})
+            sync_user_cases.delay(self)
```

Comment:
Can we add the `user.project` check here instead?

plus,
why the move to place `sync_user_cases.delay` after logging?

Reply:
I was just looking at that from code perspective where if you have two methods exiting on the first statement itself, why even call them.

>since all mobile workers do have projects

ya thats some HQ knowledge and if someone looks at the code and don't have that context, it would feel unnecessary.

Reply:
> i saw you added it back later. Still wondering if this is possible in non-test environment. If to confirm that this is just tests, you can check for user.project if its test environment and not otherwise.

It's not possible in a non-test environment, because the old behavior would have failed if that case arose, and in fact it _did_ fail in tests, but that was swallowed by the signal call.  I suppose I could put a `if unit_testing and not user.project: return` in the task.  I'll do that.